wear/src/main/kotlin/au/com/shiftyjelly/pocketcasts/wear/MainActivity.kt

```diff
@@ -59,12 +66,41 @@ fun WearApp(themeType: Theme.ThemeType) {
                 WatchListScreen(navController::navigate, it.scrollableState)
             }
 
-            composable(NowPlayingScreen.route) { viewModel ->
-                viewModel.timeTextMode = NavScaffoldViewModel.TimeTextMode.Off
+            composable(NowPlayingScreen.route) {
+                it.timeTextMode = NavScaffoldViewModel.TimeTextMode.Off
+
+                // Listen for results from streaming confirmation screen
+                navController.currentBackStackEntry?.savedStateHandle
+                    ?.getStateFlow<StreamingConfirmationScreen.Result?>(StreamingConfirmationScreen.resultKey, null)
+                    ?.collectAsStateWithLifecycle()?.value?.let { streamingConfirmationResult ->
+                        val viewModel = hiltViewModel<NowPlayingViewModel>()
+                        LaunchedEffect(streamingConfirmationResult) {
+                            viewModel.onStreamingConfirmationResult(streamingConfirmationResult)
+                            // Clear result once consumed
+                            navController.currentBackStackEntry?.savedStateHandle
+                                ?.remove<StreamingConfirmationScreen.Result?>(StreamingConfirmationScreen.resultKey)
+                        }
+                    }
```

Comment:
I don't love the boilerplate this requires to get a result from the stream warning screen via the savedStateHandle. This seemed like the best solution though since I can't use a shared view model due to the fact that I need different behavior depending on whether the stream warning is showing from the now playing screen or the episode details screen (now playing plays the currently queued episode, epsidoe details queues and plays the currently viewed episode).